src/routes/snippets.rs

```diff
@@ -81,3 +81,8 @@ pub fn create_snippet(
 
     Ok(Created(location, Some(response)))
 }
+
+#[get("/snippets/<id>")]
+pub fn get_snippet(storage: State<Box<dyn Storage>>, id: String) -> Result<JsonValue, ApiError> {
```

Comment:
I'm not sure if we want to perform some basic validation of `id` here (e.g. to verify that it's a valid slug made only of alphanumeric characters) or just let it fail to find a snippet in the database.

Reply:
I don't think it's critical to introduce such validation at this point (because we can simply fail to lookup anything in the database). I do want to check at some point some sort of declarative validation library for Rust. I think we can benefit from it in a bunch of places. :thinking:

Reply:
Sounds good!